tls/s2n_connection.c

```diff
@@ -61,16 +61,7 @@
 static int s2n_connection_new_hashes(struct s2n_connection *conn)
 {
     /* Allocate long-term memory for the Connection's hash states */
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.md5));
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.sha1));
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.sha224));
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.sha256));
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.sha384));
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.sha512));
-    POSIX_GUARD(s2n_hash_new(&conn->handshake.md5_sha1));
     POSIX_GUARD(s2n_hash_new(&conn->hash_workspace));
```

Comment:
So, why are hash_workspace, ssl3.md5, and ssl3.sha1 being left behind?

Reply:
The prf_space fields will be handled in a separate PR. They're part of a whole other thing.

Fair point on hash_workspace. It's currently a field on the connection, not the handshake, so there's no reason it currently COULDN'T be used after the handshake, but it also currently isn't. I'm torn on whether the loss in flexibility on making it explicitly a handshake value is worth the 240 bytes we save.